### PR TITLE
@orta => Use ReactiveCocoa for the DSL subspec

### LINKED ISSUE #122: Move to RAC for DSL subspec
The Aspects library we're using now doesn't support multiple injections per class hierarchy. We've been using the ReactiveCocoa solution in production for six months – may as well move over to it in master. 

Related to https://github.com/orta/ARAnalytics/issues/120

### PR DESCRIPTION
**Pull request by ashfurrow**
_Very_ much a work-in-progress. Needs:

- [x] Refactor to not be ugly
- [x] Test

Fixes #122.

### CODE REVIEW

**Comment by ashfurrow** [comment]
So refactored to look better. I moved away from `@throw`'ing exceptions, since that's an implementation detail of the Aspects library that we're not using anymore. Additionally, I've hid the implementation details of the DSL using ReactiveCocoa. 

@orta I was thinking, for anyone else who has been using the temporary fix branch, could we modify the pod spec in that branch to give them a warning or something to update, once this is merged?

**Comment by orta** [comment]
in theory that branch could have a `deprecated_by` attribute that says now they should use master

**Comment by ashfurrow** [comment]
I like it.

**Comment by ashfurrow** [comment]
Cool, this should be ready for review.

File: Bootstrap/ARAnalyticsiOSTests/ARAnalyticsDSLTests.m [hunk 1]
```diff
@@ -1,43 +1,33 @@
...
-
 #import <OCMock/OCMock.h>
 
 #import <ARAnalytics/ARAnalytics.h>
 #import <ARAnalytics/ARDSL.h>
 
 @interface TestObject : NSObject
-- (void)method1;
-- (void)method2;
+- (void)methodToBeExecuted;
```
Comment (orta) [inline Bootstrap/ARAnalyticsiOSTests/ARAnalyticsDSLTests.m]: great call here

File: Bootstrap/ARAnalyticsiOSTests/ARAnalyticsDSLTests.m [hunk 2]
```diff
@@ -70,7 +60,7 @@ - (void)method2 {}
 
             [[[analyticsMock stub] andDo:^(NSInvocation *invocation) {
                 invocationCount++;
-            }] addEventAnalyticsHooks:OCMArg.any];
+            }] addEventAnalyticsHook:OCMArg.any];
```
Comment (orta) [inline Bootstrap/ARAnalyticsiOSTests/ARAnalyticsDSLTests.m]: it feels to me like a collection of hooks, vs a single hook

**Comment by ashfurrow** [comment]
Following [this advice](https://github.com/travis-ci/travis-ci/issues/1588), I've tried to fix our current CI issues.

File: Bootstrap/ARAnalyticsiOSTests/ARAnalyticsDSLTests.m (hunk 2 above)
Comment (ashfurrow) [inline Bootstrap/ARAnalyticsiOSTests/ARAnalyticsDSLTests.m]: Tricky – it's a collection of hooks for one class. I'm fine either way.

Comment (orta) [inline Bootstrap/ARAnalyticsiOSTests/ARAnalyticsDSLTests.m]: no, I agree with the new terminology

https://github.com/artsy/eigen/blob/master/Artsy/Classes/ARAppDelegate+Analytics.m#L83 is what I though this was

**Comment by orta** [comment]
OK
This is good to merge on green, or not green.

**Comment by ashfurrow** [comment]
OK cool – I'll wait to see if there's an easy fix. Would _love_ to actually have a green PR.

**Comment by orta** [comment]
that's hard, given the lack of control in our deps.

I would be fine with it just using mixpanel for testing.

**Comment by orta** [comment]
also, once this is safely in eigen, and has had a beta run, then I'll push a 3.0.0

**Comment by ashfurrow** [comment]
Ha – that's all I've included in my own implementation! I'll commit my Podfile changes.

**Comment by ashfurrow** [comment]
Hmm. Looks like Travis is still complaining. @orta do you have advice on the GitHub key issue? It sounds like something we've discussed before ...

**Comment by ashfurrow** [comment]
@orta lawlz nevermind: https://github.com/orta/ARAnalytics/commit/b7bc6552c142880972382b345d8e0dc9068ece4b

**Pull request merged**